### PR TITLE
46 register component api call to backend

### PR DESCRIPTION
**Pull request by Magatsuken**
API call to backend
- Post request upon valid registration form
- Post request sent to correct endpoint
- console.log successful login upon successful login with status code 200
- Generates the correct error message upon failure

### CODE REVIEW

File: client/src/Login/Login.js [hunk 1]
```diff
@@ -43,9 +40,7 @@ export default function Login() {
             .then(function (response) {
```
Comment (ndohertyjr) [inline client/src/Login/Login.js]: use anonymous like this : .then(response = () => { rest of the function }

File: client/src/Login/Login.js [hunk 2]
```diff
@@ -114,7 +109,9 @@ export default function Login() {
                 setShow(true);
                 break;
             default:
-                console.log("Unhandled error code");
+                setColor(ToastConstants.color.error);
+                setMessage("Unhandled error code");
+                setShow(true);
```
Comment (ndohertyjr) [inline client/src/Login/Login.js]: should be setShowToast

File: client/src/Register/Register.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 import Button from 'react-bootstrap/Button';
 import Card from 'react-bootstrap/Card';
 import CardGroup from 'react-bootstrap/CardGroup';
 import Row from 'react-bootstrap/Row';
 import Col from 'react-bootstrap/Col';
+import Toast from '../Toast/Toast';
 
 // FIXME: Review formating of extra columns
 export default function Register() {
+    const [show, setShow] = useState(false);
```
Comment (ndohertyjr) [inline client/src/Register/Register.js]: setShowToast

File: client/src/Register/Register.js [hunk 4]
```diff
@@ -20,6 +24,34 @@ export default function Register() {
...
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({
+                Username: username,
+                Email: email,
+                Password: password
+            })
+        };
+
+        fetch('https://localhost:5001/api/auth/register', requestOptions)
```
Comment (ndohertyjr) [inline client/src/Register/Register.js]: look into configs and string interpolation for domain changes

File: client/src/Login/Login.js (hunk 1 above)
Comment (Magatsuken) [reply client/src/Login/Login.js]: Done

File: client/src/Login/Login.js (hunk 2 above)
Comment (Magatsuken) [reply client/src/Login/Login.js]: Done

File: client/src/Register/Register.js (hunk 4 above)
Comment (Magatsuken) [reply client/src/Register/Register.js]: Done

File: client/src/Register/Register.js (hunk 3 above)
Comment (Magatsuken) [reply client/src/Register/Register.js]: Done

**Review by ndohertyjr — APPROVED** [review]

**Pull request merged**